enterprise/cmd/worker/internal/codeintel/indexing/dependency_indexing_scheduler.go

```diff
@@ -88,14 +101,53 @@ func (h *dependencyIndexingSchedulerHandler) Handle(ctx context.Context, record
 			Name:    packageReference.Package.Name,
 			Version: packageReference.Package.Version,
 		}
-		if err := h.indexEnqueuer.QueueIndexesForPackage(ctx, pkg); err != nil {
-			errs = append(errs, errors.Wrap(err, "enqueuer.QueueIndexesForPackage"))
+
+		if shouldIndex {
+			if err := h.indexEnqueuer.QueueIndexesForPackage(ctx, pkg); err != nil {
+				errs = append(errs, errors.Wrap(err, "enqueuer.QueueIndexesForPackage"))
+			}
+		}
+
+		extsvcKind, ok := schemeToExternalService[packageReference.Scheme]
+		if !ok {
+			continue
+		}
+
+		if _, err := h.dbStore.InsertCloneableDependencyRepo(ctx, pkg); err != nil {
+			errs = append(errs, errors.Wrap(err, "dbstore.InsertCloneableDependencyRepos"))
+		}
+
+		if !kindExists(kinds, extsvcKind) {
+			kinds = append(kinds, extsvcKind)
+		}
+	}
+
+	// if len == 0, it will return all external services, which we definitely dont want
```

Comment:
nit: Could we have properly formatted sentences 😁 
```suggestion
	// If len == 0, it will return all external services, which we definitely don't want.
```